ls1intum/Artemis
# `Development`: Refactor long feedback association

### Checklist
#### General


#### Server

#### Changes affecting Programming Exercises

### Motivation and Context
The long feedback relation introduced in #6386 was intended to support a one-to-one lazy loading relationship. This apparently did not work.

### Description
We now model the relationship as a one-to-many in the database and ensure in the code that is is actually always a one-to-one relationship. This enables lazy loading the `LongFeedback` from the `Feedback`.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

Stacked PR on top of #6384. If you can still access the online editor, the changes from #6384 have also been successfully tested.

1. As instructor, create a test case that produces a feedback longer than 5000 characters. E.g.
   ```java
   @Test
   void longFeedback() {
       assertEquals("a".repeat(1000), "1234567890".repeat(500));
   }
   ```
2. Check the result for the template or solution repository. The feedback should contain the full message, i.e. it ends with `…1234567890>` instead of being cut off somewhere in the middle of the sequence. (Edit: Only for Jenkins setups, Bamboo never sends more than 5000 characters in the first place.)
3. Increase the `500` of the repeat to something like `2000`. Now the feedback is too long to be displayed fully in the UI, instead a download link should be shown. (Edit: Only for Jenkins setups, Bamboo never sends more than 5000 characters in the first place.)
4. In the network tab of the browser dev console: When you un-collapse the feedback message for the first time, a network request to `long-feedback` should be made. After collapsing, or even closing the feedback detail dialogue and opening it again, the request is not made again.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| Feedback.java | 97% | ✅ |
| FeedbackService.java | 100% | ✅ |


### Screenshots
n/a
